core/engine/src/vm/mod.rs

```diff
@@ -666,10 +677,18 @@ impl Context {
         self.execute_instruction(f, opcode)
     }
 
-    fn handle_error(&mut self, err: JsError) -> ControlFlow<CompletionRecord> {
+    fn handle_error(&mut self, mut err: JsError) -> ControlFlow<CompletionRecord> {
         // If we hit the execution step limit, bubble up the error to the
         // (Rust) caller instead of trying to handle as an exception.
         if !err.is_catchable() {
+            if err.backtrace.is_none() {
+                err.backtrace = Some(
+                    self.vm
+                        .shadow_stack
+                        .take(self.vm.runtime_limits.backtrace_limit(), self.vm.frame.pc),
+                );
+            }
```

Comment:
Thought: I'm wondering if this is the best place to set the backtrace info, because I don't know how this will interact with promises and generators. Did you consider setting it on the `call` and `construct` operations for function objects?

Reply:
Oh, maybe that wouldn't work because you would not get the exact instruction where the error happened. Yeah, ignore my previous comment.